services/venice-router/src/main/java/com/linkedin/venice/router/MetaDataHandler.java

```diff
@@ -274,7 +274,7 @@ public void channelRead0(ChannelHandlerContext ctx, HttpRequest req) throws IOEx
   private void handleControllerLookup(ChannelHandlerContext ctx) throws IOException {
     LeaderControllerResponse responseObject = new LeaderControllerResponse();
     responseObject.setCluster(clusterName);
-    responseObject.setUrl(routingDataRepository.getLeaderController().getUrl());
+    responseObject.setUrl(routingDataRepository.getLeaderController().getUrl(isSslToKafka));
```

Comment:
For router, should we use `sslToKafka` or `sslToStorageNodes` ?
I do not think there is a check/requirement for both to be the same.

Reply:
How to find controller's ssl property from router's code?

Reply:
Using Kafka SSL config to decide controller SSL is too confusing. Let's not update this part for now and leave a TODO message.
The correct way would be using the controller D2 service annoucement details to decide controller SSL configs, but I think it would be over-engineering for these endpoints that are hardly used.